Improve rosetta docs (#727)
o1-labs/docs2: docs/exchange-operators/rosetta/run-with-docker.mdx

---
title: Running with Docker
description: Run Mina’s implementation of Rosetta API with Docker.
keywords:
  - historical data
  - blockchain integration
  - explorer
  - wallet
  - archive node
  - graphQL
  - rosetta
  - docker
---

# Running with Docker

:::info

> There is a known issue that you can't use the docker image on the Apple Silicon chip.

Reviewer: ```suggestion
There is a known issue that you can't use the docker image on the Apple silicon chip.
```

> If you want to run Mina Rosetta on the Apple Silicon, please follow steps described in [Building and running Rosetta from source code](build-from-sources) page

Reviewer: ```suggestion
If you want to run Mina Rosetta on the Apple silicon, you can use the steps in [Building and running Rosetta from source code](build-from-sources.
```

:::

1. [Install Docker](https://www.docker.com/get-started) and check that your Docker configuration has at least 12 GB RAM (the recommended amount is 16 GB).

> 2. Check the latest release for mainnet on the official [Mina github releases page](https://github.com/MinaProtocol/mina/releases). 

Reviewer: ```suggestion
2. Check the latest release for Mainnet on the official [Mina GitHub releases](https://github.com/MinaProtocol/mina/releases) page. 
```

Reviewer: hi @qwadratic  did you know we have a word list? (we always misspell GitHub) https://github.com/o1-labs/docs2/wiki/Word-list

> 3. Go to `minaprotocol/mina-rosetta` [page on the Dockerhub](https://hub.docker.com/r/minaprotocol/mina-rosetta). Pull the image with commit hash matching to the latest mainnet release.

Reviewer: ```suggestion
3. Go to [minaprotocol/mina-rosetta](https://hub.docker.com/r/minaprotocol/mina-rosetta) ` page on the Docker Hub repository. Pull the image with the commit hash matching to the latest Mainnet release.
```

> 4. Use this image
> 
> :::note
> 
> If you want to build your own docker image, you can find more details in [Mina’s Rosetta repository](https://github.com/MinaProtocol/mina/blob/develop/src/app/rosetta/README.md). However, for most users, it’s not necessary to build your own image in order to interact with the API.
> 
> :::

Reviewer: ```suggestion
4. Use this image:

:::note

If you want to build your own docker image, you can find more details in [Mina’s Rosetta repository](https://github.com/MinaProtocol/mina/blob/develop/src/app/rosetta/README.md). However, for most users, it’s not necessary to build your own image in order to interact with the API.

:::
```

---
title: Running with Docker
description: Run Mina’s implementation of Rosetta API with Docker.
keywords:
  - historical data
  - blockchain integration
  - explorer
  - wallet
  - archive node
  - graphQL
  - rosetta
  - docker
---

# Running with Docker

:::info

There is a known issue that you can't use the docker image on the Apple silicon chip.

> To run Mina Rosetta on Apple silicon, you can use the steps in [Building and running Rosetta from source code](build-from-sources).

Reviewer 2: This is being dealt with by @joaosreis as we speak. Please consult him for an update on this issue.
